$OpenBSD: patch-lib_puppet_provider_package_openbsd_rb,v 1.1 2013/08/06 18:17:36 jasper Exp $

From a5b8b96f703d932b90ce7461b18076bb32ee85b3 Mon Sep 17 00:00:00 2001
From: Jasper Lievisse Adriaanse <jasper@humppa.nl>
Date: Tue, 23 Jul 2013 16:55:55 +0200
Subject: Enhance OpenBSD pkg.conf handling

This moves the pkg.conf parsing into it's own method which allows for better
re-use. While here, support for parsing '+=' lines has been added. This allows
for appending to PKG_PATH, instead of merely setting it with '='.

(#21930) Enchance OpenBSD pkg.conf handling

--- lib/puppet/provider/package/openbsd.rb.orig	Mon Jul 29 21:35:19 2013
+++ lib/puppet/provider/package/openbsd.rb	Mon Jul 29 21:37:15 2013
@@ -52,15 +52,18 @@ Puppet::Type.type(:package).provide :openbsd, :parent 
     [command(:pkginfo), "-a"]
   end
 
-  def install
-    should = @resource.should(:ensure)
-
+  def parse_pkgconf
     unless @resource[:source]
       if File.exist?("/etc/pkg.conf")
         File.open("/etc/pkg.conf", "rb").readlines.each do |line|
           if matchdata = line.match(/^installpath\s*=\s*(.+)\s*$/i)
             @resource[:source] = matchdata[1]
-            break
+          elsif matchdata = line.match(/^installpath\s*\+=\s*(.+)\s*$/i)
+            if @resource[:source].nil?
+              @resource[:source] = matchdata[1]
+            else
+              @resource[:source] += ":" + matchdata[1]
+            end
           end
         end
 
@@ -72,7 +75,13 @@ Puppet::Type.type(:package).provide :openbsd, :parent 
         raise Puppet::Error,
         "You must specify a package source or configure an installpath in /etc/pkg.conf"
       end
-    end
+    end  
+  end
+
+  def install
+    should = @resource.should(:ensure)
+
+    parse_pkgconf
 
     if @resource[:source][-1,1] == ::File::SEPARATOR
       e_vars = { 'PKG_PATH' => @resource[:source] }
